### PR TITLE
MAID-1919 Don't block merges when we are connected to a quorum of nodes in all sections

### PR DESCRIPTION
**Pull request by fizyk20**

### CODE REVIEW

**Comment by maidsafe-highfive** [comment]
r? @afck

(maidsafe_highfive has picked a reviewer for you, use r? to override)

File: src/peer_manager.rs [hunk 1]
```diff
@@ -391,9 +392,32 @@ impl PeerManager {
...
+    /// Checks whether we have a quorum of nodes in each section
+    fn is_merging_possible(&self) -> bool {
+        let prefixes = self.expected_peers
+            .keys()
+            .map(|x| self.routing_table.find_group_prefix(x))
+            .collect::<HashSet<_>>();
+        if prefixes.contains(&None) {
+            // we expect contacts that don't belong in any of the sections in our RT - so we have
+            // no contacts from their section
+            return false;
```
Comment (afck) [inline src/peer_manager.rs]: Actually I think we should avoid ever ending up in that situations. I'd at least log a warning here.

File: src/peer_manager.rs [hunk 2]
```diff
@@ -391,9 +392,32 @@ impl PeerManager {
...
+        }
+        // `unwrap` is safe here - we just verified that there is no `None` in the set
+        for prefix in prefixes.into_iter().map(Option::unwrap) {
```
Comment (afck) [inline src/peer_manager.rs]: It's still avoidable, and to make the actual `unwrap`s that could potentially be problematic easier to find, let's avoid them where possible. E.g. you could just `flat_map(|x| x)` instead.

File: src/peer_manager.rs (hunk 1 above)
Comment (fizyk20) [inline src/peer_manager.rs]: Isn't it possible when we for example just merged and got notified of new neighbouring sections that we should connect to, but haven't connected to anyone from them yet?

Comment (fizyk20) [inline src/peer_manager.rs]: Ah, right. I'll add a warning, then :+1:

**Pull request merged**